### ISSUE TITLE
Vaccinator app crashes every time "+" button is tapped

### ISSUE THREAD

**Issue by cagulas** [OpenSRP/opensrp-client#172]
In both Woman and Child register, the app crashes every time the "+" button is tapped to register a new client.

@maimoonak

**Comment by maimoonak** [comment]
@cagulas  Have you installed QR code reader library on device

It tries to open QR code scanner on new enrollment

**Comment by cagulas** [comment]
@maimoonak Yes, the QR code scanner app is installed. This issue is more about adding an error message or a note to the user to install the app. The app shouldn't crash when the button is pressed.

**Comment by maina** [comment]
@maimoonak I would suggest we wrap the code that starts the "com.google.zxing.client.android.SCAN" intent in a try/catch block to handle ActivityNotFoundException when the QR code scanner app isn't installed. Then in the catch block show an alert dialog asking the user to install QR code scanner app and when they click ok redirect them to the google playstore with pname filter as the zxing package:
`Uri uri = Uri.parse("market://search?q=pname:" + "com.google.zxing.client.android");
                Intent intent = new Intent(Intent.ACTION_VIEW, uri);
                    act.startActivity(intent);`
This way we avoid the crash and let the user download the QR code scanner we expect them to install.

**Comment by maimoonak** [comment]
Fixed in version 1.4.1 and would be released with pk-vaccinator-2.0.0
